Screenkey/listenkbd.py

```diff
@@ -166,14 +167,14 @@ def update_text(self, string=None):
             self.label.emit("text-changed")
 
     def key_press(self, reply):
-
         # FIXME:
         # This is not the most efficient way to detect the
         # use of sudo/gksudo but it works.
         if not self.nosudo:
-            sudo_is_running = subprocess.call(['ps', '-C', 'sudo'],
-                        stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-            if not sudo_is_running:
+            sudo_is_running = 0 == subprocess.call(
+                ['ps', '-C', 'sudo'],
+                stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            if sudo_is_running:
```

Comment:
I guess there should be `if not ...`, because ps will return 0 code if process exist.